ProcessLib/SmallDeformationWithLIE/BoundaryCondition/BoundaryConditionBuilder.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "BoundaryConditionBuilder.h"
+
+#include "MeshGeoToolsLib/BoundaryElementsSearcher.h"
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+
+#include "ProcessLib/BoundaryCondition/BoundaryConditionConfig.h"
+#include "ProcessLib/BoundaryCondition/DirichletBoundaryCondition.h"
+
+#include "NeumannBoundaryCondition.h"
+
+
+static std::vector<MeshLib::Element*> getClonedElements(
+    MeshGeoToolsLib::BoundaryElementsSearcher& boundary_element_searcher,
+    GeoLib::GeoObject const& geometry)
+{
+    std::vector<MeshLib::Element*> elements =
+        boundary_element_searcher.getBoundaryElements(geometry);
+
+    // Deep copy all the elements, because the searcher might destroy the
+    // originals. Store pointers to the copies in the elements vector (i.e.,
+    // in-place modification).
+    for (auto& e : elements)
+        e = e->clone();
+
+    return elements;
+}
+
+namespace ProcessLib
+{
+namespace SmallDeformationWithLIE
+{
+
+std::unique_ptr<BoundaryCondition> BoundaryConditionBuilder::createBoundaryCondition(
+    const BoundaryConditionConfig& config,
+    const NumLib::LocalToGlobalIndexMap& dof_table, const MeshLib::Mesh& mesh,
+    const int variable_id, const unsigned integration_order,
+    const std::vector<std::unique_ptr<ProcessLib::ParameterBase>>& parameters)
+{
+    MeshGeoToolsLib::MeshNodeSearcher& mesh_node_searcher =
+        MeshGeoToolsLib::MeshNodeSearcher::getMeshNodeSearcher(mesh);
+
+    MeshGeoToolsLib::BoundaryElementsSearcher boundary_element_searcher(
+        mesh, mesh_node_searcher);
+
+    //! \ogs_file_param{boundary_condition__type}
+    auto const type = config.config.peekConfigParameter<std::string>("type");
+
+    if (type == "Dirichlet")
```

Comment:
I think this part could be shared with the normal BC creation. `getClonedElements()` above probably, too.